test/initial_tfm.jl

```diff
@@ -0,0 +1,211 @@
+using StaticArrays, Interpolations, LinearAlgebra
+using Images, CoordinateTransformations, Rotations
+using RegisterQD
+
+g = 0.2:0.2:1.2
+gradcube = g .* reshape(g, 1, 6) .* reshape(g, 1, 1, 6)
+gradcube[1,4,2] = 0
+gradcube[4,5,2] = 0 #break up any potential rotational symmetry
+
+mxshift = (1,1,1);
+mxrot = (0.01,0.01,0.01);
+minwidth_rot = fill(0.002,3);
+
+EYE =Matrix(1.0*I, 3,3)
+
+@testset "initial_tfm improves rigid rotational alignment" begin
+    testimage1 = zeros(Float64,10,10,10)
+    testimage1 .= NaN
+    testimage1[3:8,3:8,3:8] = gradcube
+    testimage1 = centered(testimage1)
+
+    tform = RotXYZ(0.1,0.1,0.1)
+    mytform = AffineMap(tform, [0,0,0])
+    testimage2 = warp(testimage1, mytform, axes(testimage1))
+
+    # insufficient parameters + no initiat_tfm causes large misalignment
+
+    tformtest1, mm1 = qd_rigid(testimage2, testimage1, mxshift, mxrot; print_interval=typemax(Int)) #
+    @test !isapprox(1-mm1, 1)
+    @test !isapprox(tformtest1, mytform, atol = 0.1)
+
+    #initiat_tfm improves alignment
+
+    tformtest2, mm2 = qd_rigid(testimage2, testimage1, mxshift, mxrot; print_interval=typemax(Int), initial_tfm = mytform) #
+ #with the initial_tfm being the true tfm, this should give back the true rotation
+
+    @test isapprox(1-mm2, 1)
+    @test isapprox(tformtest2, mytform, atol = 0.0001)
+    @test isrotation(tformtest2.linear)
+end #Test initial_tfm improves rotational alignment for rigid
+
+@testset "initial_tfm improves rigid translational alignment" begin
+    testimage3 = zeros(10,20,10)
+    testimage3 .= NaN
+    testimage3[3:8,3:8,3:8] = gradcube
+    testimage3 = centered(testimage3)
+
+    testimage4 = zeros(10,20,10)
+    testimage4 .= NaN
+    testimage4[3:8,13:18,3:8] = gradcube
+    testimage4 = centered(testimage4)
+
+    init_tfm = AffineMap(diagm(0 => fill(1.0,3)), [0.0,10.0,0.0])
+
+    tformtest3 = nothing
+    mm3 = nothing
+
+    #no initial_tfm should fail due to total lack of overlap
+    @test try
+        tformtest3, mm3 = qd_rigid(testimage3, testimage4, mxshift, mxrot; print_interval=typemax(Int)) #
+        false
+    catch err
+        true  # this should break
+    end
+
+    tformtest4, mm4 = qd_rigid(testimage3, testimage4, mxshift, mxrot; print_interval=typemax(Int), initial_tfm = init_tfm) #
+
+    @test isapprox(1-mm4, 1) #this mismatch should be lower!
```

Comment:
```suggestion
    @test abs(mm4) < 1e-8
```
There are more places like this below, I'll stop commenting on them now.